scripts/west_commands/runners/blackmagicprobe.py

```diff
@@ -116,6 +116,7 @@ def __init__(self, cfg, gdb_serial, connect_rst=False):
         #
         # https://github.com/zephyrproject-rtos/zephyr/issues/50789
         self.elf_file = Path(cfg.elf_file).as_posix()
+        self.hex_file = Path(cfg.hex_file).as_posix()
```

Comment:
This needs an explicit check to see if `cfg.hex_file` is `None`, or it'll fail with `west build -t flash -- -DCONFIG_BUILD_OUTPUT_HEX=n`